### PR TITLE
Initialize as admin when in public mode

### PR DESCRIPTION
**Pull request by T1duS**
Fixed https://github.com/saurabhshri/ccextractor-web/issues/13

### CODE REVIEW

**Review by saurabhshri — CHANGES_REQUESTED** [review]
Thank you for the PR! :) Could you please test it again? From the first look - it doesn't look like there's any difference between both the modes right now. Could you please re-check?

**Comment by T1duS** [comment]
@saurabhshri Only the initializing part is same for both modes.

**Comment by saurabhshri** [comment]
But why? We don't want to have admin account logged in by default in `Public Mode`. 😉

**Comment by T1duS** [comment]
@saurabhshri How is this different from the issue?

**Comment by saurabhshri** [comment]
The issue is that when initialised in public mode, no admin account exist. The user needs to be granted admin role manually by editing the db entry. We want to create a default admin account in both the cases, but not login by default in public mode.

**Comment by saurabhshri** [comment]
Could you please put the common code (of admin account creation) above the if-else statements so that's it's not repetitive? Thanks!

File: run.py [hunk 1]
```diff
@@ -85,28 +85,28 @@ def init_app() -> bool:
             True if successful, False otherwise.
     """
 
-    # Setting 'Application Mode'
-    log.debug('INIT : Checking app mode.')
-    if app.config['ENABLE_LOCAL_MODE']:
-        log.debug('LOCAL MODE ENABLED')
```
Comment (saurabhshri) [inline run.py]:
```suggestion
    log.debug('INIT : Creating admin account.')
```

**Review by saurabhshri — COMMENTED** [review]
Looks good, could you please add a test to see if admin user is added successfully?

File: tests/test_running_app.py [hunk 2]
```diff
@@ -17,3 +18,10 @@ def create_app(self):
     def test_flask_application_is_up_and_running(self):
         response = app.test_client().get('/')
         self.assertEqual(response.status_code, 302)
+
+    def test_admin_user_added(self):
+        """
+        Test that an admin user is created on initializing app
+        """  
+        user = Users.query.filter(Users.account_type == AccountType.admin).first()
+        self.assertNotEqual(user, None)
```
Comment (saurabhshri) [inline tests/test_running_app.py]: Please add a newline at the end. :)

**Review by saurabhshri — CHANGES_REQUESTED** [review]
Looks good! We're close to merge. Now only the documentation needs to be edited to reflect theses changes. 👍

File: docs/installation.adoc [hunk 3]
```diff
@@ -41,7 +41,7 @@ Configured using `ENABLE_LOCAL_MODE`. Set it to`True` for _Local Mode_ and `Fals
 ==== Public Mode
 
 - To use when deploying the app on public server, where different people can access the app.
-- No default account is created.
+- Also creates an Admin User, with credentials from secret config (instance/config.py)
```
Comment (saurabhshri) [inline docs/installation.adoc]: This seems redundant. It's not needed in description of any of the modes, but needs to be mentioned separately.

**Review by saurabhshri — CHANGES_REQUESTED** [review]
Also, make sure config files have apt description too.

File: instance/config.py [hunk 4]
```diff
@@ -163,10 +163,10 @@
     Name:       ADMIN_NAME
                 ADMIN_EMAIL
                 ADMIN_PWD
-    Summary:    In case of 'local' mode, creates an admin User, with these credentials. All the operations (upload, 
-                processing, deletion etc. are done through this account. For more details about this, please refer to 
+    Summary:    Creates an admin User, with these credentials. All the operations (upload, processing,
```
Comment (saurabhshri) [inline instance/config.py]:
```suggestion
    Summary:    Creates an admin User, with these credentials. In 'local' mode, all the operations (upload, processing,
```

**Review by saurabhshri — CHANGES_REQUESTED** [review]
Looks good! :)

**Review by saurabhshri — APPROVED** [review]

**Pull request merged**